crates/router/src/connector/coinbase/transformers.rs

```diff
@@ -0,0 +1,424 @@
+use std::collections::HashMap;
+
+use serde::{Deserialize, Serialize};
+
+use crate::{
+    connector::utils::{self, AddressDetailsData, RouterData},
+    core::errors,
+    pii::Secret,
+    services,
+    types::{self, api, storage::enums},
+};
+
+#[derive(Debug, Default, Eq, PartialEq, Serialize)]
+pub struct LocalPrice {
+    pub amount: String,
+    pub currency: String,
+}
+
+#[derive(Debug, Default, Eq, PartialEq, Serialize, Deserialize)]
+pub struct Metadata {
+    pub customer_id: String,
+    pub customer_name: String,
+}
+
+#[derive(Default, Debug, Serialize, Eq, PartialEq)]
+pub struct CoinbasePaymentsRequest {
+    pub name: Secret<String>,
+    pub description: String,
+    pub pricing_type: String,
+    pub local_price: LocalPrice,
+    pub metadata: Metadata,
+    pub redirect_url: String,
+    pub cancel_url: String,
+}
+
+impl TryFrom<&types::PaymentsAuthorizeRouterData> for CoinbasePaymentsRequest {
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(_item: &types::PaymentsAuthorizeRouterData) -> Result<Self, Self::Error> {
+        get_crypto_specific_payment_data(_item)
+    }
+}
+
+// Auth Struct
+pub struct CoinbaseAuthType {
+    pub(super) api_key: String,
+}
+
+impl TryFrom<&types::ConnectorAuthType> for CoinbaseAuthType {
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(_auth_type: &types::ConnectorAuthType) -> Result<Self, Self::Error> {
+        if let types::ConnectorAuthType::HeaderKey { api_key } = _auth_type {
+            Ok(Self {
+                api_key: api_key.to_string(),
+            })
+        } else {
+            Err(errors::ConnectorError::FailedToObtainAuthType.into())
+        }
+        // Err(errors::ConnectorError::NotImplemented("try_from ConnectorAuthType".to_string()).into())
+    }
+}
+// PaymentsResponse
+#[derive(Debug, Clone, Default, Serialize, Deserialize, PartialEq)]
+#[serde(rename_all = "UPPERCASE")]
+pub enum CoinbasePaymentStatus {
+    New,
+    #[default]
+    Pending,
+    Completed,
+    Expired,
+    Unresolved,
+    Resolved,
+    Cancelled,
+    #[serde(rename = "PENDING REFUND")]
+    PendingRefund,
+    Refunded,
+}
+
+impl From<CoinbasePaymentStatus> for enums::AttemptStatus {
+    fn from(item: CoinbasePaymentStatus) -> Self {
+        match item {
+            CoinbasePaymentStatus::Completed | CoinbasePaymentStatus::Resolved => Self::Charged,
+            CoinbasePaymentStatus::Expired => Self::Failure,
+            CoinbasePaymentStatus::New => Self::AuthenticationPending,
+            CoinbasePaymentStatus::Unresolved => Self::Unresolved,
+            _ => Self::Pending,
+        }
+    }
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
+#[serde(rename_all = "UPPERCASE")]
+pub enum UnResolvedContext {
+    Underpaid,
+    Overpaid,
+    Delayed,
+}
+
+impl From<UnResolvedContext> for String {
```

Comment:
check, using strum crate can we automatically derive this, Strum::Display can derive impl for Enum to String automatically. How to enable casing we need to check.